llvm/test/CodeGen/AMDGPU/dagcombine-fma-fmad.ll

```diff
@@ -39,7 +39,7 @@ define amdgpu_ps float @_amdgpu_ps_main() #0 {
 ; GFX10-NEXT:    v_sub_f32_e32 v8, s0, v1
 ; GFX10-NEXT:    v_fma_f32 v7, -s2, v6, s6
 ; GFX10-NEXT:    v_fma_f32 v5, v6, v5, 1.0
-; GFX10-NEXT:    v_mad_f32 v10, s2, v6, v2
+; GFX10-NEXT:    v_fma_f32 v10, s2, v6, v2
```

Comment:
Some of these test diffs don't make sense to me. Here we traded mad for fma (which probably shouldn't have happened), and below the opposite (traded fma for mad)

Reply:
Changing fma to mad was a bug in the patch, now fixed.

Reply:
> Here we traded mad for fma (which probably shouldn't have happened)

I think this happens because we only form FMAD post-legalization. Any folding that happens earlier will use FMA.

Reply:
I think the restriction to only introduce it post legalization was dumb, I didn't repeat that in GlobalISel. It's easiest to just treat it consistently